packages/thirdweb/src/react/core/machines/paymentMachine.ts

```diff
@@ -70,7 +70,7 @@ export interface PaymentMachineContext {
 /**
  * Events that can be sent to the payment machine
  */
-type PaymentMachineEvent =
+export type PaymentMachineEvent = { state?: PaymentMachineState } & ( // Current state override to get out of overly strict switch statement hell
```

Comment:
oof this is generally a code smell, breaks the logical ordering of states because now you can go from any state to any state.

i would recommend trying to fit this new code path into a regular event that gets processed like any other event